ui-core/src/participant/landing/ConfiguredMedia.tsx

```diff
@@ -0,0 +1,88 @@
+import classNames from 'classnames'
+import React, { CSSProperties } from 'react'
+import { useApiContext } from '../../participant/ApiProvider'
+
+import { requireOptionalString, requirePlainObject, requireOptionalNumber }
+  from '../../participant/util/validationUtils'
+
+export type VideoConfig = {
+  videoLink: string
+  alt?: string
+  className?: string
+  style?: CSSProperties
+}
+
+export type ImageConfig = {
+  cleanFileName: string
+  version: number
+  alt?: string
+  className?: string
+  style?: CSSProperties
+}
+
+export type MediaConfig = ImageConfig | VideoConfig
+
+// TODO: Add JSDoc
+// eslint-disable-next-line jsdoc/require-jsdoc
+export const validateMediaConfig = (imageConfig: unknown): MediaConfig => {
+  const message = 'Invalid image config'
+  const config = requirePlainObject(imageConfig, message)
+
+  const cleanFileName = requireOptionalString(config, 'cleanFileName', message)
+  const version = requireOptionalNumber(config, 'version', message)
+  const alt = requireOptionalString(config, 'alt', message)
+  const className = requireOptionalString(config, 'className', message)
+  const videoLink = requireOptionalString(config, 'videoLink', message)
+  // Only validate that style is an object. React will handle invalid keys.
+  const style = config.style ? requirePlainObject(config.style, `${message}: Invalid style`) : undefined
+
+  return {
+    cleanFileName,
+    version,
+    alt,
+    className,
+    style,
+    videoLink
+  } as MediaConfig
+}
+
+type ConfiguredImageProps = {
+  media: MediaConfig
+  className?: string
+  style?: CSSProperties
+}
+
+/** renders an image that is part of a SiteContent spec */
+export default function ConfiguredMedia(props: ConfiguredImageProps) {
+  const { media, className, style } = props
+  const { getImageUrl } = useApiContext()
+  if ((media as VideoConfig).videoLink) {
+    const videoLinkMedia = media as VideoConfig
+    const videoAllowed = isVideoLinkAllowed(videoLinkMedia.videoLink)
+    return <div style={{ ...style, ...media.style }}
+      className={classNames('configured-image', className, media.className)}>
+      {videoAllowed && <iframe src={videoLinkMedia.videoLink} frameBorder="0" allowFullScreen={true}
+        data-testid="media-iframe"></iframe> }
```

Comment:
I noticed that depending on where I placed the image on the page, the full screen button was visible but unreachable. Could just be an issue with interactive content that users will work around, or perhaps something that could be helped with a larger border?

Reply:
yeah, that's something to keep an eye on.  In the meantime, users can add a className of "p-{x}" to add padding as they need